### DISCUSSION TITLE
GROUP_BY algorithm optimized for sorted data

### DISCUSSION THREAD

**Discussion by ankravch** [duckdb/duckdb#11841] [General]
I wonder if duckdb has GROUP_BY that is optimized for sorted data.  
For example,  ```SELECT max(B) from read_parquet('sorted_by_A.parquet') GROUP BY A;```
where sorted_by_A.parquet has 1B rows.   Using HASH_GROUP_BY wouldn't be optimal in this case.

**Comment by Maxxen** [comment]
No, such an aggregate operator is not currently present, although this is something we want to support in the future.

**Comment by hawkfish** [comment]
This is part of #8444
